pkg/api/persistentvolumeclaim/util_test.go

```diff
@@ -384,82 +385,217 @@ func TestDataSourceRef(t *testing.T) {
-			feature:  true,
-			pvc:      withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
-			oldPVC:   getPVC(),
-			expected: withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			name:                                "for:newPVC=hasResizeStatus,oldPVC=hasResizeStatus,featuregate=true; should keep field",
+			enableRecoverVolumeExpansionFailure: true,
+			enableVolumeAttributesClass:         false,
+			pvc:                                 withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			oldPVC:                              withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			expected:                            withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
 		},
 		{
-			name:     "for:newPVC=hasResizeStatus,oldPVC=hasResizeStatus,featuregate=true; should keep field",
-			feature:  true,
-			pvc:      withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
-			oldPVC:   withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
-			expected: withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			name:                                "for:newPVC=hasResizeStatus,oldPVC=hasResizeStatus,featuregate=false; should keep field",
+			enableRecoverVolumeExpansionFailure: false,
+			enableVolumeAttributesClass:         false,
+			pvc:                                 withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			oldPVC:                              withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			expected:                            withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
 		},
 		{
-			name:     "for:newPVC=hasResizeStatus,oldPVC=hasResizeStatus,featuregate=false; should keep field",
-			feature:  false,
-			pvc:      withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
-			oldPVC:   withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
-			expected: withResizeStatus(core.PersistentVolumeClaimNodeResizeFailed),
+			name:                                "for:newPVC=hasVolumeAttributeClass,oldPVC=nil, featuregate=false should drop field",
+			enableRecoverVolumeExpansionFailure: false,
+			enableVolumeAttributesClass:         false,
+			pvc:                                 withVolumeAttributesClassName("foo"),
+			oldPVC:                              nil,
+			expected:                            getPVC(),
+		},
+		{
+			name:                                "for:newPVC=hasVolumeAttributeClass,oldPVC=doesnot,featuregate=true; should keep field",
+			enableRecoverVolumeExpansionFailure: false,
+			enableVolumeAttributesClass:         true,
+			pvc:                                 withVolumeAttributesClassName("foo"),
+			oldPVC:                              getPVC(),
+			expected:                            withVolumeAttributesClassName("foo"),
+		},
+		{
+			name:                                "for:newPVC=hasVolumeAttributeClass,oldPVC=hasVolumeAttributeClass,featuregate=true; should keep field",
```

Comment:
When featuregate=true, which feature gate does it refer to?  Does it mean if one of the feature gates enableRecoverVolumeExpansionFailure or enableVolumeAttributesClass is enabled, this featuregate =true? In either case, plase add comments to clarify.